src/jemalloc.c

```diff
@@ -865,6 +865,23 @@ malloc_conf_multi_sizes_next(const char **slab_size_segment_cur,
 	return false;
 }
 
+static void
+malloc_conf_format_error(const char *msg, const char *begin, const char *end) {
```

Comment:
Can't we just leverage the existing `printf` infrastructure and do something like:
```c
size_t len = end - begin + 1;
len = len > BUFERROR_BUF ? BUFERROR_BUF : len;
malloc_printf("%s%.*s\n", msg, (int) len, begin);
```
Instead of hand-formatting everything?

Reply:
For these warnings we usually go through `malloc_write` so that they can be redirected; see here: https://github.com/jemalloc/jemalloc/blob/dev/src/malloc_io.c#L81

Reply:
In that case, we could `malloc_snprintf` into a local buffer and then `malloc_write` that buffer out, if you feel like this is better than hand-formatting.

Reply:
Actually you were right. malloc_printf takes care of the malloc_message as well. Let me switch to that.